### PR TITLE
Allow copying also the content of the config map

### PR DESCRIPTION
**Pull request by davidfestal**
Signed-off-by: David Festal <dfestal@redhat.com>

### What does this PR do?

This PR allows correctly propagating up to the workspace, the content of the trusted CA bundle configured 

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18245

### How to test this PR?

- Install Che
- Create a `my-ca-certs` config map with 1 key name `my.crt` and an untrusted certificate as value
- Set the `spec.server.ServerTrustStoreConfigMapName` of the `CheCluster` custom resource to the name of the `my-ca-certs` config map
- Wait for the Che server restart
- Start a workspace
- Check that the namespace of the workspace contains a `ca-certs` config map that contains a `ca-bundle.crt` key (with OpenShift-injected certificates) **and** the `my.crt` key with your untrusted certificate as a value.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java [hunk 1]
```diff
@@ -67,9 +79,22 @@ public void provision(KubernetesEnvironment k8sEnv, OpenShiftProject project)
     if (!trustedStoreInitialized) {
       return;
     }
+    ConfigMap configMap =
+        clientFactory
+            .createOC()
+            .configMaps()
+            .inNamespace(cheInstallationLocation.getInstallationLocationNamespace())
```
Comment (mshaposhnik) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: Can we do this installation namespace evaluation just once in constructor?

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java [hunk 2]
```diff
@@ -43,17 +47,25 @@
...
   private final boolean trustedStoreInitialized;
+  private final String caBundleConfigMap;
   private final String configMapName;
   private final Map<String, String> configMapLabelKeyValue;
+  private final CheInstallationLocation cheInstallationLocation;
+  private final OpenShiftClientFactory clientFactory;
 
   @Inject
   public Openshift4TrustedCAProvisioner(
+      CheInstallationLocation cheInstallationLocation,
```
Comment (mshaposhnik) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: Usualy we put class type parameters after the `@Named` strings.

**Review by mshaposhnik — APPROVED** [review]

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java (hunk 1 above)
Comment (davidfestal) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: Well, then we should enable throwing an `InfrastructureException` from the constructor

Comment (mshaposhnik) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: Its not a problem, i mean throwing exc from constructor, we already did that in some other classes.

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java (hunk 2 above)
Comment (davidfestal) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: Fixed in commit https://github.com/eclipse/che/pull/18264/commits/2d143ee3ed3e9643917e3844503597ada90bf41d

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java (hunk 1 above)
Comment (davidfestal) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/Openshift4TrustedCAProvisioner.java]: OK, let me do the change then.

**Comment by davidfestal** [comment]
@mshaposhnik @skabashnyuk This PR should probably be cherry-picked to the `main` branch.

Is this something platform team plan to integrate into next Sprint ? 

cc @l0rd

**Pull request merged**